front/src/modules/trainschedule/components/Timetable/PacedTrain/hooks/useOccurrences.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from 'react';
+
+import dayjs from 'dayjs';
+
+import { type Duration } from 'utils/duration';
+
+import type { PacedTrainWithResult } from '../../types';
+import type { Occurrence } from '../OccurrenceItem';
+
+type OccurrencesState = {
+  occurrences: Occurrence[];
+  occurrencesCount: number;
+};
+
+const useOccurrences = (pacedTrain: PacedTrainWithResult, stepDuration: Duration) => {
+  const [occurrencesState, setOccurrencesState] = useState<OccurrencesState>({
+    occurrences: [],
+    occurrencesCount: 0,
+  });
+
+  const computeOccurrenceName = (trainName: string, index: number): string => {
+    const endByNumber = /\b\w+\s\d+$/;
+
+    if (endByNumber.test(trainName)) {
+      const endOfTrainName = Number(trainName.split(' ').pop());
+      return `${trainName.replace(/\s\d+$/, '')} ${endOfTrainName + 2 * index}`;
+    }
+    if (!Number.isNaN(+trainName)) {
+      return `${+trainName + 2 * index}`;
+    }
+    return `${trainName} ${2 * index + 1}`;
+  };
+
+  useEffect(() => {
+    const computeOccurrences = (): OccurrencesState => {
+      const occurrencesCount = Math.floor(pacedTrain.paced.duration.ms / pacedTrain.paced.step.ms);
+      const computedOccurrences = [];
+      for (let i = 0; i < occurrencesCount; i += 1) {
+        const startTime = dayjs(pacedTrain.startTime)
+          .add(i * stepDuration.ms, 'ms')
+          .toDate();
+        const arrivalTime = dayjs(pacedTrain.startTime)
+          .add((i + 1) * stepDuration.ms, 'ms')
+          .toDate();
+        computedOccurrences.push({
+          id: `occurrence-${i}-${pacedTrain.id}`,
+          trainName: computeOccurrenceName(pacedTrain.trainName, i),
+          rollingStock: pacedTrain.rollingStock,
+          startTime,
+          arrivalTime,
+        });
+      }
+      return { occurrencesCount, occurrences: computedOccurrences };
+    };
+
+    setOccurrencesState(computeOccurrences());
+  }, [pacedTrain]);
```

Comment:
No need to pass `stepDuration` in useEffect dependencies ?
```suggestion
  }, [pacedTrain, stepDuration]);
```

Reply:
We remove the second hook parameters so we don't need to watch it anymore.